src/scroll-timeline-css.js

```diff
@@ -144,36 +144,20 @@ export function initCSSPolyfill() {
 
   initMutationObserver();
 
-  // Cache all Proxy Animations
-  let proxyAnimations = new WeakMap();
-
   // We are not wrapping capturing 'animationstart' by a 'load' event,
   // because we may lose some of the 'animationstart' events by the time 'load' is completed.
   window.addEventListener('animationstart', (evt) => {
     evt.target.getAnimations().filter(anim => anim.animationName === evt.animationName).forEach(anim => {
-      // Create a per-element cache
-      if (!proxyAnimations.has(evt.target)) {
-        proxyAnimations.set(evt.target, new Map());
-      }
-      const elementProxyAnimations = proxyAnimations.get(evt.target);
-
-      // Store Proxy Animation in the cache
-      if (!elementProxyAnimations.has(anim.animationName)) {
-        const result = createScrollTimeline(anim, anim.animationName, evt.target);
-        if (result && result.timeline && anim.timeline != result.timeline) {
-          elementProxyAnimations.set(anim.animationName, new ProxyAnimation(anim, result.timeline, result.animOptions));
-        } else {
-          elementProxyAnimations.set(anim.animationName, null);
-        }
-      }
-      
-      // Get Proxy Animation from cache
-      const proxyAnimation = elementProxyAnimations.get(anim.animationName);
-
-      // Swap the original animation with the proxied one
-      if (proxyAnimation !== null) {
+      const result = createScrollTimeline(anim, anim.animationName, evt.target);
```

Comment:
result may be null.

Reply:
+1 on `result` possibly being `null`

Reply:
> result may be null.

Thanks, done!

> Canceling and restarting a CSS animation will unconditionally create a new ScrollTimeline. Possibly not a big performance hit, but it looked like this was one of the things the previous cache PR was trying to avoid.

I think createScrollTimeline is pretty inexpensive. Scroll driven animations themselves will generally start once and run forever. Non-scroll driven animations should quickly be determined not to match any of the rules in getAnimationTimeline (and we didn't used to cache misses anyways). There are other easy optimizations we could do here if it became a bottleneck.